src/Umbraco.Web/PublishedContentExtensions.cs

```diff
@@ -158,10 +149,14 @@ public static object Value(this IPublishedContent content, string alias, bool re
         /// <para>If eg a numeric property wants to default to 0 when value source is empty, this has to be done in the converter.</para>
         /// <para>The alias is case-insensitive.</para>
         /// </remarks>
-        public static object Value(this IPublishedContent content, string alias, bool recurse, object defaultValue)
+        public static object Value(this IPublishedContent content, string alias,  string culture = null, string segment = null,  object defaultValue = default, bool recurse = false)
         {
-            var property = content.GetProperty(alias, recurse);
-            return property == null || property.HasValue() == false ? defaultValue : property.GetValue();
+            var property = content.GetProperty(alias);
+
+            if (property != null && property.HasValue(culture, segment))

```

Comment:
This check always fails therefore you never get back a value for a culture variant, HasValue doesn't contextualize anything which i think is part of the problem.

Reply:
contextualization should happen anytime a public method on a property is invoked, which is the case for `GetValue()`, `GetXPathValue()`... but yeh, `HasValue()` is also a public method that needs to contextualize - fixing